pyhoma/models.py

```diff
@@ -184,51 +202,44 @@ def __len__(self) -> int:
 class Command(dict):  # type: ignore
     """Represents an TaHoma Command."""
 
-    __slots__ = (
-        "name",
-        "parameters",
-    )
-
     def __init__(self, name: str, parameters: str | None = None, **_: Any):
         self.name = name
         self.parameters = parameters
         dict.__init__(self, name=name, parameters=parameters)
 
 
 # pylint: disable-msg=too-many-locals
+@attr.s(auto_attribs=True, init=False, slots=True, kw_only=True)
 class Event:
-    __slots__ = (
-        "timestamp",
-        "name",
-        "gateway_id",
-        "exec_id",
-        "deviceurl",
-        "device_states",
-        "old_state",
-        "new_state",
-        "owner_key",
-        "setupoid",
-        "owner_key",
-        "type",
-        "sub_type",
-        "time_to_next_state",
-        "failed_commands",
-        "failure_type_code",
-        "failure_type",
-        "condition_groupoid",
-        "placeoid",
-        "label",
-        "metadata",
-        "camera_id",
-        "deleted_raw_devices_count",
-        "protocol_type",
-    )
+    timestamp: int
+    name: EventName
+    setupoid: str | None = attr.ib(repr=offuscate_id, default=None)
```

Comment:
Does this mean that this id will always we obfuscated? I would love to have this in the logging, however I think that we should not obfuscate by default. 

Can we just set obfuscation in our logging? We don't want to lose any information or context in our application itself.

Reply:
If you directly access the value, it will not be obfuscated. It's only when `__repr__` is called.

> repr (a bool or a callable to use a custom function.) – Include this attribute in the generated __repr__ method. If True, include the attribute; if False, omit it. By default, the built-in repr() function is used. To override how the attribute value is formatted, pass a callable that takes a single value and returns a string. Note that the resulting string is used as-is, i.e. it will be used directly instead of calling repr() (the default).

Reply:
That's great! Do you think we should document this in our README? (which has 0 documentation at the moment by the way 😀).

Don't think that people will be converting an object to a string for other purposes than logging though, let's just merge it.

Reply:
well, we will see if one day someone complain about this :)